src/visor/terravisor/services/kernel/helios_sched.c

```
@@ -360,7 +366,7 @@ void _helios_scheduler_despatch(void)
 	else
 	{
 		/* A System Error case */
-		arch_panic_handler();
+		_helios_sched_panic("Presched Callback not provided!");
```

may be use status_t values?

---

We do not want to return a error status here to the user. It shall directly panic.

---

no i was thinking something like https://github.com/VisorFolks/cyancore/blob/stable/src/driver/driver.c#L205

---

Updated